### PR TITLE
Account for properties and additionalProperties in allOf

### PR DESCRIPTION
**Pull request by msutkowski**
Hi there! We've had a handful of issues reporting inaccurate type generation for users of [RTK Query Code Generator](https://github.com/reduxjs/redux-toolkit/tree/master/packages/rtk-query-codegen-openapi) which leverages oazapfts.

I made a quick reproduction comparing oazapfts output to a user-provided schema to various other generators here: https://github.com/msutkowski/oazapfts-typesecript-codegen-repro. You can see that other generators will produce a type like:

`type Dog = Animal & { uniqueDogProp?: string | null }`. 

With the current version of oazapfts, we were generating

`type Dog = Animal`

This PR addresses that by accounting for `properties` and `additionalProperties` when handling the `schema.allOf` case. The diff in the output looks like this against the repro above:

image

Referenced issues with reports:

- https://github.com/reduxjs/redux-toolkit/issues/2002
- https://github.com/reduxjs/redux-toolkit/issues/1940

### CODE REVIEW

**Comment by msutkowski** [comment]
Ah, should've checked before I opened this - but it's a duplicate of #120

**Comment by msutkowski** [comment]
Just noting for anyone else that might be following this on RTKQ's behalf - we're going to start [maintaining a patched version](https://github.com/rtk-incubator/oazapfts) of this for issues that impact our users directly.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (jonkoops) [inline .prettierrc:1]: Could we remove the Prettier config from this PR? Seems out of scope for the changes here.

**Review by jonkoops — CHANGES_REQUESTED** [review]
Other changes look good. If we can get rid of the Prettier config I think we can merge this.

File: .prettierrc (hunk 1 above)
Comment (msutkowski) [reply .prettierrc]: removed.

**Comment by Xiphe** [comment]
oh, were the changes here also part of https://github.com/cellular/oazapfts/pull/234 ?
